### ISSUE TITLE
Cypress cy.request function sends the POST request to the server but Cypress responds with TIMEOUT

### ISSUE THREAD

**Issue by kapalkat** [cypress-io/cypress#1889]
### Current behavior:
Cypress cy.request function sends the POST request to the server but Cypress responds with TIMEOUT.
From my app perspective all works fine: the POST call is processed correctly; 204 HTTP success code is sent back.

Doesn't matter how I change the timeout within the cy.request() function it's always the same.

```
CypressError: cy.request() failed trying to load:

http://localhost:81/operations/api/trains/planning-session/new

We attempted to make an http request to this URL but the request failed without a response.

We received this error at the network level:

  > Error: ESOCKETTIMEDOUT

-----------------------------------------------------------

The request we sent was:

Method: POST
URL: http://localhost:81/operations/api/trains/planning-session/new

-----------------------------------------------------------

Common situations why this would fail:
  - you don't have internet access
  - you forgot to run / boot your web server
  - your web server isn't accessible
  - you have weird network configuration settings on your computer

The stack trace for this error is:

RequestError: Error: ESOCKETTIMEDOUT
    at new RequestError (/Users/kapalkat/projects/frontend/node_modules/cypress/dist/Cypress.app/Contents/Resources/app/packages/server/node_modules/request-promise-core/lib/errors.js:14:15)
    at Request.plumbing.callback (/Users/kapalkat/projects/frontend/node_modules/cypress/dist/Cypress.app/Contents/Resources/app/packages/server/node_modules/request-promise-core/lib/plumbing.js:87:29)
    at Request.RP$callback [as _callback] (/Users/kapalkat/projects/frontend/node_modules/cypress/dist/Cypress.app/Contents/Resources/app/packages/server/node_modules/request-promise-core/lib/plumbing.js:46:31)
    at self.callback (/Users/kapalkat/projects/frontend/node_modules/cypress/dist/Cypress.app/Contents/Resources/app/packages/server/node_modules/request/request.js:186:22)
    at emitOne (events.js:115:13)
    at Request.emit (events.js:210:7)
    at ClientRequest.<anonymous> (/Users/kapalkat/projects/frontend/node_modules/cypress/dist/Cypress.app/Contents/Resources/app/packages/server/node_modules/request/request.js:781:16)
    at Object.onceWrapper (events.js:314:30)
    at emitNone (events.js:105:13)
    at ClientRequest.emit (events.js:207:7)
    at Socket.emitTimeout (_http_client.js:722:34)
    at Object.onceWrapper (events.js:314:30)
    at emitNone (events.js:105:13)
    at Socket.emit (events.js:207:7)
    at Socket._onTimeout (net.js:402:8)
    at ontimeout (timers.js:469:11)
    at tryOnTimeout (timers.js:304:5)
    at Timer.listOnTimeout (timers.js:264:5)
```
### Desired behavior:
The POST is send and cypress doesn't respond with any error message
### Steps to reproduce:
I have created following cy.request():
```js
function createTrainsPlanningSession(name, ignoreIfExists = false) {
    const IGNORED_ERROR_MSG = 'Planning session with that name already exists'

    console.debug("Add new planning session with name: " + name)
    cy.request({
        method: 'POST',
        url: '/operations/api/trains/planning-session/new',
        auth: {
            bearer: window.sessionStorage.getItem('auth_token')
        },
        body: {
            newPlanningSession: name,
            basePlanningSession: " "
        },
        timeout: 120000,
        failOnStatusCode: false
    }).then((resp) => {
        if (!(resp.isOkStatusCode || (ignoreIfExists && resp.body === IGNORED_ERROR_MSG))) {
            const EXCEPTION_MSG = 'Exception when creating planning session'
            console.error(EXCEPTION_MSG + ':\n' + JSON.stringify(resp))
            throw new Error(EXCEPTION_MSG + '.\nCode: ' + JSON.stringify(resp.status) + '\nMessage:' + JSON.stringify(resp.body))
        }
    })
```
Then I am calling the function like so:
```js
describe('TA-OPL-01: Planning Sessions: Trains', function () {
    const PATH_PLANNING_TRAINS = '/operations/#/menu/planning/trains';

    before(function () {
        cy.login()
    })

    beforeEach(function () {
        cy.setUpAuth()
    })

    it('TA-OPL-01-01: Check Drag and Drop', function () {
        const PLANNING_SESSION_NAME = 'TEST_SESSION15'

        createTrainsPlanningSession(PLANNING_SESSION_NAME, true)

        cy.visit(PATH_PLANNING_TRAINS + "/list")
        cy.waitForLoad()
    })
})
```

### Versions
checked with versions 2.1.0 and on new version: 3.0.1
System Version: macOS 10.13.5 (17F77)
browser: Chrome

**Comment by jennifer-shehane** [comment]
- How long does the request to this endpoint general take to respond outside of Cypress?

[This issue](https://github.com/cypress-io/cypress/issues/943#issuecomment-347069798) may be relevant as well

**Comment by jennifer-shehane** [comment]
I investigated [this previously for `cy.visit()`](https://github.com/cypress-io/cypress/issues/1657#issuecomment-386068530) and this looks to be the same thing that is happening with your `cy.request()`. 

Network connections eventually time out at around ~30 seconds by Cypress/the underlying operating system, which then causes the cy.request to fail. This is something we can "kind of" tweak in Cypress - basically we can up the timeout for ESOCKETTIMEOUT but only up to the maximum the operating system allows.

Why is the request taking 45 seconds? Are you waiting for your server to start? If so, there are some [better alternatives](https://github.com/cypress-io/cypress/issues/1657#issuecomment-386068530) than waiting in the `cy.request()` itself.

**Comment by kapalkat** [comment]
My app binaries are stored locally for development and testing but the DB is stored far away in Customer infrastructure (VPN involved also). I can't get the db locally as it contains confidential data.
Is there any work around?
________________________________

**Comment by kapalkat** [comment]
I have checked other REST clients to check if I am able to send the request and get the correct response and I have no problem with that.
I have checked POSTMAN and SOAPUI.

SOAP UI response example:

HTTP/1.1 204 
Server: nginx/1.12.2
Date: Fri, 08 Jun 2018 08:03:35 GMT
Connection: keep-alive
X-Application-Context: application:dev:8081

response time 39236ms

POSTMAN response example is the same except it took 45 seconds.

I have checked Cypress on both: MacOS and Windows and the request is failing on both systems.

I was also able to test the request with Cypress remotely on Customer infrastructure where backend and DB are close each other with internal networking and this is working fine.
